### DISCUSSION TITLE
How to make a Rule trigger A0 above and below a certain threshold ?

### DISCUSSION THREAD

**Discussion by SammyGirlie** [arendst/Tasmota#13380] [Q&A] [answered]
Hi,
I have an analog sensor (A0) and want this sensor to trigger a MQTT message whenver its values crosses the limit of 500 - no matter in what direction.
- The default value is around 750 ... as soon as it reaches 500 it should publish MQTT - peaking up to approx. 40 without another publish
- As soon as it drops down to the default - crossing the 500 again, it should again publish MQTT.

I have written a rule for this, but the rule publishes all the time, flooding my MQTT server, no matter if the value crosses 500 or not.

The role looks like this:

Rule1 
ON ANALOG#A0 DO Var1 %value% ENDON 
ON analog#a0<500 DO publish tele/%topic%/SENSOR {"ANALOG":{"A0":%Var1%}} ENDON 
ON analog#a0>500 DO publish tele/%topic%/SENSOR {"ANALOG":{"A0":%Var1%}} ENDON

Do you have an idea how this can be fixed ?
BR
Sammy

**Comment by barbudor** [comment, marked as answer]
https://tasmota.github.io/docs/Rules/#usage-of-one-shot-once

And you can avoid copying the value into %var1%. Just use %value% in the publish payload

**Reply by SammyGirlie** [reply]
But I read in another thread that it is not possible to access the %value% variable more than once within one rule ?

**Reply by barbudor** [reply]
Where ?
I see no reason. The %value% in `ON analog#a0<500 DO publish tele/%topic%/SENSOR {"ANALOG":{"A0":%value%}} ENDON`  would be the exact value that has triggered the rule.
With your current version, it will be the one before as %xx% substitutions are performed **before** execution

**Reply by SammyGirlie** [reply]
Ok, it works with %value% only, but the "one-shot-once" rule does not apply:

ON analog#a0<500 DO publish tele/%topic%/SENSOR 1 ENDON
ON analog#a0>500 DO publish tele/%topic%/SENSOR 0 ENDON

Now it pushes just 0 and 1, but nevertheless several times a second :)

**Reply by barbudor** [reply]
> but the "one-shot-once" rule does not apply

What do you mean ? Have you enabled the ONCE mode ?
`Rule1 5` ?

**Reply by SammyGirlie** [reply]
No I did not ...

But now I did and now it works perfectly fine for me 👯‍♀️ 
Thanks a lot for your great support !!

**Reply by SammyGirlie** [reply]
Rule to trigger whenever you cross the "border":

Rule1 
ON analog#a0<500 DO publish   tele/%topic%/SENSOR 1 ENDON 
ON analog#a0>500 DO publish   tele/%topic%/SENSOR 0 ENDON

Option to make it a One-Shot Rule:
Rule1 5

**Reply by barbudor** [reply]
Was just the 1st line of my 1st answer ... 🤣
